src/main/java/net/sf/jabref/logic/importer/fetcher/MrDLibFetcher.java

```diff
@@ -0,0 +1,145 @@
+
+/**
+ *  This class is responible to get the recommendations from MDL
+ *
+ */
+public class MrDLibFetcher implements EntryBasedFetcher {
+
+    private static final String NAME = "MDL_FETCHER";
+    private static final Log LOGGER = LogFactory.getLog(MrDLibFetcher.class);
+    private final String LANGUAGE;
+    private final String VERSION;
+    private final JabRefPreferences prefs = JabRefPreferences.getInstance();
+
+    public MrDLibFetcher() {
+        LANGUAGE = prefs.get(JabRefPreferences.LANGUAGE);
+        VERSION = new BuildInfo().getVersion().getFullVersion();
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+
+    @Override
+    public List<BibEntry> performSearch(BibEntry entry) throws FetcherException {
+        String response = makeServerRequest(entry.getLatexFreeField("title").get());
+        MrDLibImporter importer = new MrDLibImporter();
+        ParserResult parserResult = new ParserResult();
+        try {
+            if (importer.isRecognizedFormat(new BufferedReader(new StringReader(response)))) {
+                parserResult = importer.importDatabase(new BufferedReader(new StringReader(response)));
+            } else {
+                // For displaying An ErrorMessage
+                BibEntry errorBibEntry = new BibEntry();
+                errorBibEntry.setField("html_representation",
+                        Localization.lang("Error_while_fetching_from_%0", "Mr.DLib"));
+                BibDatabase errorBibDataBase = new BibDatabase();
+                errorBibDataBase.insertEntry(errorBibEntry);
+                parserResult = new ParserResult(errorBibDataBase);
+            }
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        return parserResult.getDatabase().getEntries();
+    }
+
+    /**
+     * Contact the server with the title of the selected item
+     * @param query: The query holds the title of the selected entry. Used to make a query to the MDL Server
+     * @return Returns the server response. This is an XML document as a String.
+     * @throws FetcherException
+     */
+    private String makeServerRequest(String query) throws FetcherException {
+        query = constructQuery(query);
+        String response = "";
+        SSLContext sslContext = null;
+
+        try {
+            sslContext = SSLContexts.custom().loadTrustMaterial(null, new TrustSelfSignedStrategy()).build();
+        } catch (KeyManagementException | NoSuchAlgorithmException | KeyStoreException e1) {
+            LOGGER.error(e1.getMessage(), e1);
+        }
+
+        SSLConnectionSocketFactory sslSocketFacktory = new SSLConnectionSocketFactory(sslContext);
+
+        try (CloseableHttpClient httpclient = HttpClients.custom().setSSLSocketFactory(sslSocketFacktory).build()) {
+            Unirest.setHttpClient(httpclient);
+            try {
+                response = Unirest.get(query).asString().getBody();
```

Comment:
Didn't know that you already discussed the Unirest thing. Sorry.
Nonetheless, it might be a good idea to use the URLDownloader especially since it already handles SSL certificates and thus might resolve the issue Jörg was mentioning.